src/backends/duckdb.rs

```diff
@@ -25,37 +25,12 @@ pub fn query(
     format: &OutputFormat,
     writer: &OutputWriter,
 ) -> Result<()> {
-    let mut query = query.to_string();
-    if query.starts_with("prql ") {
-        // prepend CTEs for the source aliases
-        let mut lines: Vec<String> = query.split("\n").map(|s| s.to_string()).collect();
-        debug!("sources = {sources:?}");
-        for (alias, source) in sources.iter() {
-            debug!("alias = {alias:?}; source = {source:?}");
-            // Needs the _{}_ on the LHS for _{}_.*
-            lines.insert(
-                1,
-                format!("table {alias} = (from __{alias}__=__file_{alias}__)"),
-            );
-        }
-        query = lines.join("\n");
-        debug!("query = {query:?}");
-    }
+    let sql_query = if query.starts_with("prql ") {
+        let mut stmts = parse(query)?;
 
-    // compile the PRQL to SQL
-    let mut sql = get_sql_from_query(&query)?;
-    debug!(
-        "sql = {:?}",
-        sql.split_whitespace().collect::<Vec<&str>>().join(" ")
-    );
-
-    if query.starts_with("prql ") {
-        // replace the table placeholders again
-        for (alias, source) in sources.iter() {
-            debug!("alias = {alias:?}; source = {source:?}");
```

Comment:
I need to look over these changes more and try them out. I know what I had was quite hacky but I think there was a reason for it.

Reply:
You had something like this:

```
prql_source
  .split_by_lines()
  .insert(
     1, "table source_name = (from a_name = a_key)"
  )
  .join('\n')
  .compile_prql()
  .replace('a_key', "read_csv('source_file.csv')")
```

I've changed it to:
```
let pl_stmts = parse(prql_source)
pl_stmts.insert(1, parse("let source_name = s'''read_csv('source_file.csv')'''"))
compile(pl_stmts)
```